### PR TITLE
Oauth2 Support

### PR DESCRIPTION
**Pull request by karen-avetisyan-mc**
Adding OAuth 2.0 code

### CODE REVIEW

File: src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        if (basePath.isEmpty()){
+            throw new IllegalArgumentException("basePath in application.properties is empty");
+        }
+        this.basePath = basePath;
+        if (signingKeyContainer.isEmpty()){
+            throw new IllegalArgumentException("pkcs12KeyFile in application-oauth2.properties is empty");
+        }
+        this.signingKeyContainer = signingKeyContainer;
+        if (clientId.isEmpty()){
+            throw new IllegalArgumentException("client id  in application-oauth2.properties is empty");
```
Comment (NehaSony) [inline src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java:53]: Should be `client id in application-oauth2.properties is empty` -> has two spaces

File: src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        if(keyId.isEmpty()) {
+            throw new IllegalArgumentException("keyId in application-oauth2.properties is empty");
+        }
+
+        this.keyId = keyId;
+        if (scope.isEmpty()) {
+            throw new IllegalArgumentException("scope in application-oauth2.properties is empty");
+        }
+        this.scope = scope;
+        if (dpopKeyType.isEmpty()) {
```
Comment (NehaSony) [inline src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java:83]: This can be simplified to :
`this.dpopKeyType = (dpopKeyType == null || dpopKeyType.isEmpty()) ? "PS256" : dpopKeyType;`

File: src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java
Comment (NehaSony) [inline src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java:1]: Consider adding unit tests to verify that the constructor throws exceptions for missing/empty values, ensuring default values are reflected, and the client can create the object correctly
